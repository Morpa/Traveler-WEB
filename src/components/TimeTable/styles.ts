import styled, { css } from 'styled-components'

export const Wrapper = styled.article`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    padding: 1rem;
    width: 10rem;
    height: 8.4rem;
    background: ${theme.colors.primary};
    border-radius: 0.8rem;
    border: 1px solid ${theme.colors.blueLow};

    h3 {
      margin-top: ${theme.spacings.xxsmall};
      color: ${theme.colors.titleColor};
    }

    h4 {
      font-weight: ${theme.font.light};
      color: ${theme.colors.textColor};
    }
  `}
`
