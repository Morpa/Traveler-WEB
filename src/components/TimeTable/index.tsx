import * as S from './styles'

export type TimeTableProps = {
  weekDay: string
  hours: string
}

const TimeTable = ({ weekDay, hours }: TimeTableProps) => (
  <S.Wrapper>
    <h4>{weekDay}</h4>
    <h3>{hours}</h3>
  </S.Wrapper>
)

export default TimeTable
