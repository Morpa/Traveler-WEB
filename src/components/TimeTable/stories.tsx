import { Story, Meta } from '@storybook/react/types-6-0'
import TimeTable, { TimeTableProps } from '.'

export default {
  title: 'TimeTable',
  component: TimeTable,
  args: {
    weekDay: 'Domingo',
    hours: '8h - 17h'
  }
} as Meta

export const Default: Story<TimeTableProps> = (args) => <TimeTable {...args} />
