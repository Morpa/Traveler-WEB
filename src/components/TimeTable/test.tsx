import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import TimeTable from '.'

const props = {
  weekDay: 'Domingo',
  hours: '8h - 17h'
}

describe('<TimeTable />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<TimeTable {...props} />)

    expect(
      screen.getByRole('heading', { name: /domingo/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /8h - 17h/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
