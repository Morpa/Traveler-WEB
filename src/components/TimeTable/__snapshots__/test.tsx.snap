// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeTable /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  width: 10rem;
  height: 8.4rem;
  background: #FFFFFF;
  border-radius: 0.8rem;
  border: 1px solid #DDE9F0;
}

.c0 h3 {
  margin-top: 0.8rem;
  color: #123952;
}

.c0 h4 {
  font-weight: 300;
  color: #617480;
}

<article
  class="c0"
>
  <h4>
    Domingo
  </h4>
  <h3>
    8h - 17h
  </h3>
</article>
`
